src_c/display.c

```diff
@@ -1112,6 +1112,12 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
                  this invalidates the display surface*/
                 SDL_SetWindowTitle(win, title);
                 SDL_SetWindowSize(win, w_1, h_1);
+                if (0 !=
+                    SDL_SetWindowFullscreen(
+                        win, sdl_flags & (SDL_WINDOW_FULLSCREEN |
+                                          SDL_WINDOW_FULLSCREEN_DESKTOP))) {
+                    return RAISE(pgExc_SDLError, SDL_GetError());
+                }
```

Comment:
To avoid potential future regressions, a comment above this explaining it has to come before setting other flags like resize/bordered/hidden etc would be nice